app/src/main/java/com/ece493/group5/adjustableaudio/ui/media_player/MediaPlayerFragment.java

```diff
@@ -258,16 +268,9 @@ private void checkStoragePermission()
 
         if (!hasReadPermission || !hasWritePermission)
         {
-            disableMediaControls();
-            addMediaButton.setOnClickListener(new View.OnClickListener()
-            {
-                @Override
-                public void onClick(View view)
-                {
-                    showStoragePermissionsDialog();
-                }
-            });
+            return false;
         }
+        return true;

```

Comment:
You can turn this into a single return statement:

```java
return hasReadPermission && hasWritePermission;
```

Looks cleaner in my opinion, but not required.